servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpCookiePair.java

```diff
@@ -87,7 +87,7 @@ public static HttpCookiePair parseCookiePair(final CharSequence sequence, int na
 
     static HttpCookiePair parseCookiePair0(final CharSequence sequence, int nameStart, int nameLength, int valueEnd) {
         final int valueStart = nameStart + nameLength + 1;
-        if (valueEnd - 1 < valueStart) {
+        if (valueEnd <= valueStart || valueStart < 0) {
             throw new IllegalArgumentException("unexpected format of cookie pair, empty value");
```

Comment:
This exception is a bit confusing and doesn't tell what was wrong with the arguments. Consider validating some passed params independently and thow if any of them are incorrect. For example, if `nameStart < 0` or `nameLength` is not positive.

Reply:
I'll adjust the exception message to include more context.